lib/HLSL/HLOperationLower.cpp

```diff
@@ -6871,34 +6861,57 @@ void GenerateStructBufSt(Value *handle, Value *bufIdx, Value *offset,
 }
 
 
-static Value* TranslateStructBufVecLd(Type* VecEltTy, unsigned ElemCount,
+static Value* TranslateStructOrRawBufVecLd(Type* VecEltTy, unsigned ElemCount,
   IRBuilder<>& Builder, Value* handle, hlsl::OP* OP, Value* status,
-  Value* bufIdx, Value* baseOffset, const DataLayout& DL, std::vector<Value*> &bufLds, bool isScalarTy) {
+  Value* bufIdx, Value* baseOffset, const DataLayout& DL,
+  std::vector<Value*> &bufLds, DXIL::ResourceKind kind, bool isScalarTy) {
+
+  DXASSERT_NOMSG(kind == DXIL::ResourceKind::StructuredBuffer ||
+    kind == DXIL::ResourceKind::RawBuffer);
+
   unsigned  EltSize = DL.getTypeAllocSize(VecEltTy);
+  // If RawBuffer load of 64-bit value, don't set alignment to 8,
+  // since buffer alignment isn't known to be anything over 4.
+  if (kind == HLResource::Kind::RawBuffer && EltSize > 4) {
+    EltSize = 4;
```

Comment:
I don't like that we are changing EltSize here, this doesn't seem right.  We need to keep track of alignment separately.  In the future, we should be adding something to allow a base alignment promise in HLSL for the buffer declaration.  This would allow higher alignment for these cases.